src/utils.zig

```diff
@@ -1530,6 +1530,228 @@ pub const BinaryWriter = struct {
 	}
 };
 
+const ReadWriteTest = struct {
+	fn init() void {
+		main.initThreadLocals();
+	}
+	fn deinit() void {
+		main.deinitThreadLocals();
+	}
+	fn getWriter() BinaryWriter {
+		return .init(main.stackAllocator, .big);
+	}
+	fn getReader(data: []const u8) BinaryReader {
+		return .init(data, .big);
+	}
+	fn testInt(comptime intT: type, expected: intT) !void {
+		var writer = getWriter();
+		defer writer.deinit();
+		writer.writeInt(intT, expected);
+
```

Comment:
I would like you to also add an assert for the sizes, the size of the buffer after writing should always be `@ceilDiv(@bitSizeOf(intT), 8)`

Reply:
Ah right, they never added that. There's std.math.divCeil though, and I'd appreciate you using it since it clearer conveys the intention here.

Reply:
Done